src/main.js

```diff
@@ -82,6 +88,17 @@ module.exports = {
         }
       }
     }))
+    this.subscriptions.add(
+    atom.config.observe('linter-eslint.localNodeModules', localNodeModules => {
+      if (localNodeModules) {
+        atom.config.set('linter-eslint.useGlobalEslint', false)
+      }
+    }))
+    this.subscriptions.add(atom.config.observe('linter-eslint.useGlobalEslint', useGlobalEslint => {
+      if (useGlobalEslint) {
+        atom.config.set('linter-eslint.localNodeModules', '')
```

Comment:
Permanently changing the value of the setting based on whether or not the Use Global setting is checked is a bad idea, you should simply not use it in the worker if this is enabled if that was your intent.